### ISSUE TITLE
Partially written comment lost when switching between panes

### ISSUE THREAD

**Issue by bhackett1024** [RecordReplay/devtools#5437]
I started writing a comment while paused at a location in a recording, and then switched to the pause pane so that I could get some information I needed to continue writing the comment.  When I switched back to the comment pane to finish the comment, the text I had already written was gone.

**Comment by markerikson** [comment]
Without actually looking at the source yet, my immediate guess would be that the comment component keeps the input text in local component state.  When you switch to another tab, the comment component is unmounted, and so all of its state goes away.

The first solutions I can think of are:

- Persist the input text in `localStorage`
- Copy the data over to Redux

Either of these would probably only be done if the component is about to unmount and there is unsaved text in the input.  I would strongly advise _against_ attempting to tie the input itself directly to Redux (ie, don't dispatch an action on every keystroke).

**Comment by jasonLaster** [comment]
Thanks for reporting. I believe this was a regression since the intent is to preserve the comment. 

This should be easy to record with Replay and see what is going wrong. Also once we have a fix, we can ask QAWolf to write a test for it

**Comment by jcmorrow** [comment]
I think we used to store the `pendingComment` in redux, but that concept sort of broke down once we allowed for more than one editor. There's a piece of work that would help with this which is "Allow for comments to be uploaded with their ID already set". That way we could generate UUID's on the front-end, use those to persist partially written content, and then restore them when remounting the editors. This would also help with comment flickering. Currently we do [a silly amount of work] (https://github.com/RecordReplay/devtools/commit/b49c40557118e3a9a97adf090ea2067dd47c3b49) to try and uniquely identify comments without being able to know what UUID will be sent back from the server after they are persisted.
